vertica_python/vertica/cursor.py

```diff
@@ -58,8 +58,7 @@ def execute(self, operation, parameters=None):
 
         self.rowcount = 0
         if self.last_execution:
-            # ToDo: can just empty the message buffer in connection if easy to do
-            self.connection.reset_connection()
```

Comment:
I think your PR is awesome for now; I'll think about the cases for a connection reset and submit another PR if I think of anything.